### PR TITLE
Render GraphTrack curves as step functions

### PR DESCRIPTION
**Pull request by pierricgimmig**
- labels don't snap to points anymore
- clamp label x position
- fix division by zero bug

### CODE REVIEW

File: OrbitGl/GraphTrack.cpp [hunk 1]
```diff
@@ -53,8 +53,9 @@ void GraphTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, PickingM
       float x1 = time_graph_->GetWorldFromTick(time);
       float y0 = base_y + static_cast<float>(last_normalized_value) * size_[1];
       float y1 = base_y + static_cast<float>(normalized_value) * size_[1];
-      batcher->AddLine(Vec2(x0, y0), Vec2(x1, y1), graph_z, kLineColor);
-      DrawSquareDot(batcher, Vec2(x1, y1), kDotRadius, dot_z, kDotColor);
+      batcher->AddLine(Vec2(x0, y0), Vec2(x1, y0), graph_z, kLineColor);
+      batcher->AddLine(Vec2(x1, y0), Vec2(x1, y1), graph_z, kLineColor);
+      DrawSquareDot(batcher, Vec2(x1, y0), kDotRadius, dot_z, kDotColor);
```
Comment (karupayun) [inline OrbitGl/GraphTrack.cpp:58]: The square dot has to be in (x1, y1).

**Review by karupayun — APPROVED** [review]
Thanks for fixing this Pierric!

I leaved a comment and I think that we want to continue the blue lines at the end of the capture to be consistent.

Screen Shot 2020-10-29 at 14 06 19

File: OrbitGl/GraphTrack.cpp (hunk 1 above)
Comment (pierricgimmig) [reply OrbitGl/GraphTrack.cpp:58]: yeah you're right! thanks!

**Comment by pierricgimmig** [comment]
> I think that we want to continue the blue lines at the end of the capture to be consistent.

I agree, done. Thanks Pablo!

**Pull request merged**